int32[] flat_buffer
int32 height
int32 width
int32 depth
